pkg/sql/create_index.go

```diff
@@ -423,6 +423,17 @@ func replaceExpressionElemsWithVirtualCols(
 			}
 
 			if !isInverted && !colinfo.ColumnTypeIsIndexable(typ) {
+				if colinfo.ColumnTypeIsInvertedIndexable(typ) {
+					return errors.WithHint(
+						pgerror.Newf(
+							pgcode.InvalidTableDefinition,
+							"index element %s of type %s is not indexable. an inverted index can be created instead",
+							elem.Expr.String(),
+							typ.Name(),
+						),
+						"see the documentation for more information about inverted indexes: "+docs.URL("inverted-indexes.html"),
```

Comment:
You can update the hint to suggest an inverted index after following @postamar's suggestion above. Something like: `you may want to create an inverted index instead. See the documentation for inverted indexes: ...`